api/envoy/api/v2/eds.proto

```diff
@@ -51,6 +51,9 @@ message ClusterLoadAssignment {
   // List of endpoints to load balance to.
   repeated endpoint.LocalityLbEndpoints endpoints = 2 [(gogoproto.nullable) = false];
 
+  // List of named endpoints that can be referenced multiple times in LocalityLbEndpoints.
+  repeated endpoint.Endpoint named_endpoints = 5;
```

Comment:
Cool, this looks simpler. Could even minimize the PR further by making this a map with key being host name string and value being the endpoint. WDYT?

Reply:
Works for me. Like how it enforces the key uniqueness